src/devices/Mhz19b/Mhz19b.cs

```diff
@@ -0,0 +1,218 @@
+
+            return (concentration, validity);
+        }
+
+        /// <summary>
+        /// Initiates a zero point calibration.
+        /// The sensor doesn't respond anything, so this is fire and forget.
+        /// </summary>
+        /// <returns>true, if the command could be send</returns>
+        public bool ZeroPointCalibration() => SendRequest(CreateRequest(Command.CalibrateZeroPoint));
+
+        /// <summary>
+        /// Initiate a span point calibration.
+        /// The sensor doesn't respond anything, so this is fire and forget.
+        /// </summary>
+        /// <param name="span">span value, e.g. 2000[ppm]</param>
+        /// <returns>true, if the command could be send</returns>///
+        public bool SpanPointCalibration(int span)
+        {
+            var request = CreateRequest(Command.CalibrateSpanPoint);
+            // set span in request, c. f. datasheet rev. 1.0, pg. 8 for details
+            request[(int)MessageFormat.DataHigh] = (byte)(span / 256);
+            request[(int)MessageFormat.DataLow] = (byte)(span % 256);
+
+            return SendRequest(request);
+        }
+
+        /// <summary>
+        /// Switches the autmatic baseline correction on and off.
+        /// The sensor doesn't respond anything, so this is fire and forget.
+        /// </summary>
+        /// <param name="state">State of automatic correction</param>
+        /// <returns>true, if the command could be send</returns>
+        public bool AutomaticBaselineCorrection(AbmState state)
+        {
+            var request = CreateRequest(Command.AutoCalibrationSwitch);
+            // set on/off state in request, c. f. datasheet rev. 1.0, pg. 8 for details
+            request[(int)MessageFormat.DataHigh] = (byte)state;
+
+            return SendRequest(request);
+        }
+
+        /// <summary>
+        /// Set the sensor detection range.
+        /// The sensor doesn't respond anything, so this is fire and forget
+        /// </summary>
+        /// <param name="detectionRange">Detection range of the sensor</param>
+        /// <returns>true, if the command could be send</returns>
+        public bool SensorDetectionRange(DetectionRange detectionRange)
+        {
+            var request = CreateRequest(Command.DetectionRangeSetting);
+            // set detection range in request, c. f. datasheet rev. 1.0, pg. 8 for details
+            request[(int)MessageFormat.DataHigh] = (byte)((int)detectionRange / 256);
+            request[(int)MessageFormat.DataLow] = (byte)((int)detectionRange % 256);
+
+            return SendRequest(request);
+        }
+
+        private bool SendRequest(byte[] request)
+        {
+            bool validity = false;
+
+            request[(int)MessageFormat.Checksum] = Checksum(request);
+
+            try
+            {
+                _serialPort.Open();
+                _serialPort.Write(request, 0, request.Length);
+                validity = true;
+            }
+            finally
```

Comment:
Would be better to raise an IOException rather than returning an issue? If there is an issue with the serial port, none of the other commands will work anyway.

Reply:
Whether I use an exception or a return code for reporting the status is a matter of taste or project coding guideline.
Usually I prefer the return code, but if preferred by the iot-project I could change this. Please advice.

Reply:
See previous comment. When related to a serial port or I2C or SPI, then it's better to raise an IOException.
If it's related to a bad reading, to non possible data, then no exception.
And good point, I'll add this in the documentation as well so it will be more clear and consistent

Reply:
Makes sense to me. I'll change this to an exception.

Reply:
Done.

Reply:
Why do you open and close the port each time? I guess that increases the risk of getting garbage data in the communication stream.